jdk17-unit-tests/src/test/java/com/uber/nullaway/jdk17/NullAwaySwitchTests.java

```diff
@@ -156,4 +156,20 @@ public void testSwitchExprUnbox() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void testSwitchExprLambda() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "SwitchExpr.java",
+            "package com.uber;",
+            "import java.util.function.Function;",
+            "class SwitchExpr {",
+            "  int i;",
+            "  public void testSwitchExprLambda() {",
+            "    Function<SwitchExpr,Object> f = (s) -> switch (s.i) { case 3, 4, 5 -> new Object(); default -> \"hello\"; };",
```

Comment:
I suppose the only thing to test here is whether a crash can happen or not, since we can't really add any interesting nullability test here until we support nullability of generics to some degree...

Reply:
Yes, exactly.  Sorry, I had set this to automerge, but I'll add a comment in a follow-up PR